third-party/realdds/src/dds-device-broadcaster.cpp

```diff
@@ -4,45 +4,45 @@
 #include <realdds/dds-device-broadcaster.h>
 
 #include <realdds/dds-participant.h>
+#include <realdds/dds-publisher.h>
+#include <realdds/dds-topic-writer.h>
 #include <realdds/dds-utilities.h>
 #include <realdds/dds-guid.h>
-#include <realdds/topics/device-info/device-info-msg.h>
-#include <realdds/topics/device-info/deviceInfoPubSubTypes.h>
+#include <realdds/topics/flexible/flexible-msg.h>
+#include <realdds/topics/flexible/flexiblePubSubTypes.h>
 
 #include <librealsense2/utilities/easylogging/easyloggingpp.h>
-#include <fastdds/dds/domain/DomainParticipantFactory.hpp>
-#include <fastdds/dds/domain/DomainParticipant.hpp>
-#include <fastdds/dds/domain/DomainParticipantListener.hpp>
-#include <fastdds/dds/publisher/Publisher.hpp>
-#include <fastdds/dds/publisher/DataWriter.hpp>
-#include <fastdds/dds/topic/TypeSupport.hpp>
-#include <fastdds/dds/publisher/qos/PublisherQos.hpp>
-#include <fastdds/rtps/participant/ParticipantDiscoveryInfo.h>
+#include <librealsense2/utilities/json.h>
 
 #include <iostream>
 
 using namespace eprosima::fastdds::dds;
 using namespace realdds;
 
 // We want to know when readers join our topic
-class dds_device_broadcaster::dds_client_listener : public eprosima::fastdds::dds::DataWriterListener
+class dds_device_broadcaster::dds_client_listener : public realdds::dds_topic_writer //dds_topic_writer is a listener
 {
 public:
-    dds_client_listener( dds_device_broadcaster * owner )
-        : eprosima::fastdds::dds::DataWriterListener()
+    dds_client_listener( std::shared_ptr< dds_topic > const & topic,
+                         std::shared_ptr< dds_publisher > const & publisher,
+                         dds_device_broadcaster * owner )
+        : dds_topic_writer( topic, publisher )
```

Comment:
The client listener is also publishing data?
Which data?

Reply:
In the old implementation there were 2 objects. a) A writer of type `eprosima::fastdds::dds::DataWriter` b) The listener which inherited `eprosima::fastdds::dds::DataWriterListener`. 

I changed this to use `dds_topic_writer` that is a listener that contains a writer. No need for a separate writer object.
The name might be optimized a bit to convey the abilities better. Using the same object for listening and sending `device-info` topic when a reader is matched.

Reply:
👍